Major cleanup

lucasdf opened the pull request:
Converted from CoffeeScript to JavaScript
Configured linter and fixed lint issues
Upgraded  to API v2
Added specs
Fixed some messages not being displayed
Fixed `lintsOnChange` was set to `true` although the package was not checking the content on change

Arcanemagus commented on lib/init.js:
```diff
@@ -0,0 +1,105 @@
+            const patterns = [
+              {
+                // regex for "ERROR: ENV invalid format ENV_VARIABLE on line 2"
+                regex: /(WARN|ERROR):(.*) on line (\d+)/,
```
Shouldn't this be `/(WARN|ERROR): (.*) on line (\d+)/` to remove the initial space from the message?

Arcanemagus commented on lib/init.js:
```diff
@@ -0,0 +1,105 @@
+              },
+              {
+                // regex for "ERROR: Multiple CMD instructions found, only line 3 will take effect"
+                regex: /(WARN|ERROR):(.*), only line (\d+)/,
```
Same here, remove the initial space from the captured message.

Arcanemagus commented on lib/init.js:
```diff
@@ -0,0 +1,105 @@
+
+              messages.push({
+                severity: match.severity === 'WARN' ? 'warning' : 'error',
+                excerpt: match.excerpt.trim(),
```
Fixing the regex above should remove the need for `.trim()` here.

Arcanemagus commented on lib/init.js:
```diff
@@ -0,0 +1,105 @@
+              {
+                // regex for "ERROR: ENV invalid format ENV_VARIABLE on line 2"
+                regex: /(WARN|ERROR):(.*) on line (\d+)/,
+                cb: m => ({ lineNo: m[3], excerpt: m[2], severity: m[1] }),
```
Since we're doing arithmetic on it later, do a `Number.parseInt(m[3], 10)` on the `lineNo`.

Arcanemagus commented on lib/init.js:
```diff
@@ -0,0 +1,105 @@
+              {
+                // regex for "ERROR: Multiple CMD instructions found, only line 3 will take effect"
+                regex: /(WARN|ERROR):(.*), only line (\d+)/,
+                cb: m => ({ lineNo: m[3], excerpt: m[2], severity: m[1] }),
```
Same here, do a `Number.parseInt(m[3], 10)` on the `lineNo`.

Arcanemagus commented on spec/linter-docker-spec.js:
```diff
@@ -0,0 +1,89 @@
+    // https://github.com/AtomLinter/Meta/issues/15
+    const activationPromise = atom.packages.activatePackage('linter-docker');
+
+    // await atom.packages.activatePackage('language-python');
```
Since this package isn't using `activationHooks`, this whole block could be replaced with just this:
```js
await atom.packages.activatePackage('linter-docker');
```

If you do want to use the `activationHook` compatible method just in case it gets implemented later then the right way to do it would be to activate the associated language **and** open a file of that language. Since opening a file adds ~1 second to each specs time, it's probably best to just go with the above version for now.

lucasdf commented:
@johnwebbcole thanks for reviewing this! I added instructions in the README for running the tests and linting the project.

@Arcanemagus I fixed those issues!

Arcanemagus commented:
Looks like this should be a patch release since it was never actually linting changes in the first place 😛.

lucasdf commented:
This should close both #10 and #5

Arcanemagus commented:
Published in v0.2.1. 🎉

lucasdf commented:
Great!

lucasdf commented:
Now we may setup CircleCI!
